qcodes/utils/magic.py

```diff
@@ -145,44 +152,88 @@ def measurement(self, line, cell=None):
                 else:
                     # variable exists in user namespace
                     val = eval(key, self.shell.user_ns)
-                msmt_name = msmt_name.replace(f'{{{key}}}', str(val), 1)
+                msmt_name = msmt_name.replace(f"{{{key}}}", str(val), 1)
 
         # Add dataset
         contents += f"{data_name} = {loop_name}.get_data_set(name='{msmt_name}')"
-        if 's' not in options:
+        if "s" not in options:
             contents += f"\nprint({data_name})"
 
-        for line in lines[k+1:]:
-            contents += '\n' + line
+        for line in lines[k + 1 :]:
+            contents += "\n" + line
 
-        if 'r' in options:
-            contents += f'\n{loop_name}.run(thread=True)'
+        if "r" in options:
+            contents += f"\n{loop_name}.run(thread=True)"
 
-        if 'p' in options:
+        if "p" in options:
             print(contents)
 
-        if 'x' not in options:
+        if "x" not in options:
             # Execute contents
             self.shell.run_cell(contents, store_history=True, silent=True)
 
+    @cell_magic
+    def new_job(self, line, cell):
+        """Run code in cell in a separate thread.
+        This is especially useful for running measurements in a separate thread.
+        The thread is an IPython job. It can be accessed via
+        `qcodes.active_job`, and the job manager via `qcodes.job_manager`.
+
+        Args:
+            line: Optional name for the thread. Default is '_measurement'.
+                If a different name is used, it will not update
+                `qcodes.active_job`.
+            cell: Code to be executed in a separate thread
+
+        Raises:
+            RuntimeError if a measurement thread already exists
+
+        Note:
+            Only one job can be active at a time.
+            A side effect is that the code in the cell is first converted to a
+                function whose name is _measurement
```

Comment:
_measurement should actually be `thread_name` which comes from `line`. This side effect only exists in the context of this function call though, right?